src/librustc_back/target/armv7neon_linux_androideabi.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use LinkerFlavor;
+use target::{Target, TargetOptions, TargetResult};
+
+// See https://developer.android.com/ndk/guides/abis.html#v7a
+// for target ABI requirements.
+
+pub fn target() -> TargetResult {
+    let mut base = super::android_base::opts();
+    base.features = "+v7,+thumb-mode,+thumb2,+vfp3,+d16,+neon".to_string();
```

Comment:
`+v7` and `+thumb2` are redundant given the `armv7` in the llvm_target. `+d16` should also be removed given NEON requires 32 registers (see [ARM®
 Architecture Reference Manual](https://static.docs.arm.com/ddi0406/cd/DDI0406C_d_armv7ar_arm.pdf) A1-33).